src/plugins/elasticsearch/lib/map_uri.js

```diff
@@ -1,6 +1,23 @@
 import querystring from 'querystring';
 import { resolve } from 'url';
+import _ from 'lodash';
+
 module.exports = function mapUri(server, prefix) {
+
+  const normalizeHeader = function (header) {
+    return header.trim().toLowerCase();
+  };
+
+  const filterHeaders = function (originalHeaders) {
+    const originalHeadersNormalized = _.mapKeys(originalHeaders, function (headerValue, headerName) {
```

Comment:
BTW, I could replace `function (headerValue, headerName) { ... }` over here with `_.rearg(normalizeHeader, 1, 0)` and it would do the same thing. I chose not to use `_.rearg`, however, because I think it lowers readability of this code. I'm curious what the team norms are around this sort of decision: use more verbose code (like I have) to preserve readability and clear intent vs. use library functions to keep code succint. /cc @rashidkpc